### ISSUE TITLE
"Overzoom" problem when using custom map server (Tileserver-GL).

### ISSUE THREAD

**Issue by TwinTurbine** [dceejay/RedMap#264]
Hi!

Having ditched the idea of using PMTiles natively in Worldmap the Tileserver-GL works very well serving MBTiles. However there seems to be missing a setting similar to the "maxDataZoom" that the documentation describes when using PMTiles.

The issue I'm experiencing now is that when I zoom beyond the maximum zoom level that is supported by the MBTile served by Tileserver-GL, Worldmap will first zoom one step further in the already served raster map. About one second later Worldmap will send a GET request for the next zoom level which the server will respond to with a 404. This in turn will remove the overlay map completely, exposing the base map underneath. So for one second it looks good, and the next second it's gone.

I might of course have missed something but is it possible to stop Worldmap from requesting a zoom level that is known to be unavailable and just continue to zoom in the last served raster when this is the case? I think such a setting needs to be supplied on a per-map-basis as different overlay maps might have different zoom limitations.

Kind regards,

**Comment by dceejay** [comment]
Can you not add a maxZoom property when adding the layer ? as per docs
https://github.com/dceejay/RedMap?tab=readme-ov-file#to-add-a-new-base-layer

**Comment by TwinTurbine** [comment]
Well, I've tried that but it doesn't seem to have that effect. The maxZoom property just seems to disable the overlay beyond the set maxZoom level altogether. From the documentation:

`The maxDataZoom should match the maximum zoom level in you pmtiles file(s) - whereas the maxZoom is the leaflet maximum zoom level you want to support.`

It seems to me that a maxDataZoom property would be useful in the case with MBtiles/Tileserver as well so that Worldmap doesn't request a zoom level that will render a 404 response, thus removing the map.

**Comment by TwinTurbine** [comment]
Found it! After having tried almost everything that Internet had to offer, including the `maxzoom` parameter for sources in the config.json for the TileServer-GL (which did nothing), I directed my attention to the Leaflet documentation.

And there it was, the `maxNativeZoom` parameter. This effectively stops Leaflet from requesting tiles beyond that zoom level,  instead zooming in the last served zoom level. Maybe this parameter could be useful to add to the Worldmap documentation for custom base/overlays?
